### PR TITLE
WT-8908 Expand s_all to check return values are in parentheses

### PR DESCRIPTION
**Pull request by ruby-oujo**
* Added a check to `s_all` in `s_style` to check that return values are in brackets.
  * Modified the regex to ensure that comments and strings with 'return' will not be picked up. 
  * Output with filename, line number and expression. 
  * Tested with edge cases. 
  
* Modified code in files that have missed parentheses.

### CODE REVIEW

File: dist/s_style [hunk 1]
```diff
@@ -46,6 +46,13 @@ else
 		if egrep -r -n '\b[a-z]+[A-Z]' $f | egrep -v ':[     ]+\*|"|UNCHECKED_STRING'; then
 			echo "$f: Styling requires variables that use underscores to separate parts of a name instead of camel casing.";
 		fi
+		egrep -r -n '^[^*/"]*[[:space:]]*return [^(]' $f > $t;
```
Comment (korteland) [inline dist/s_style:49]: Nitpick: there should be a line break here to separate it from the logically separate task of picking up camel case errors. The comments around here could also a little clearer now that you've added some more functionality, e.g.:

```
# Ignore styling errors of external libraries
if expr "$f" : 'src/' > /dev/null &&
  ...

  # Camel case style is not allowed.
  if egrep -r -n '\b[a-z]+[A-Z]' $f | egrep -v ':[     ]+\*|"|UNCHECKED_STRING'; then
  ...

  # Return values must be wrapped in parentheses.
  egrep -r -n '^[^*/"]*[[:space:]]*return [^(]' $f > $t;
  ...
```

**Review by korteland — COMMENTED** [review]
One nitpick, LGTM otherwise. Nice work :-)

File: dist/s_style (hunk 1 above)
Comment (ruby-oujo) [reply dist/s_style:49]: You make a very good point! I forgot to add commenting,  I will adjust this now.

**Review by keithbostic — APPROVED** [review]
Nice! I didn't think this was going to be doable.

**Review by korteland — APPROVED** [review]
Thanks for fixing the whitespace thing!

**Pull request merged**